plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/startup/PluginCoreEnforcementActivity.kt

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.startup
+
+import com.intellij.ide.ApplicationInitializedListener
+import com.intellij.ide.plugins.PluginEnabler
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.openapi.application.ex.ApplicationManagerEx
+import com.intellij.openapi.extensions.PluginId
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import kotlinx.coroutines.CoroutineScope
+import org.slf4j.LoggerFactory
+
+class PluginCoreEnforcementActivity : ApplicationInitializedListener {
+
+    @Suppress("LazyLog")
+    override suspend fun execute(asyncScope: CoroutineScope) {
+        // can't reference anything in core
+        val log = LoggerFactory.getLogger(this::class.java)
+
+        val coreId = PluginId.getId("aws.toolkit.core")
+        val coreDescriptor = PluginManagerCore.getPlugin(coreId)
+        if (coreDescriptor != null) {
+            // already installed
+            return
+        }
+
```

Comment:
In case core is installed but disabled we will need to explicitly enable for them.